### PR TITLE
backup report: show negative compression

### PR DESCRIPTION
**Pull request by pstorz**
#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

File: core/src/dird/backup.cc [hunk 1]
```diff
@@ -949,12 +949,12 @@ void GenerateBackupSummary(JobControlRecord *jcr, ClientDbRecord *cr, int msg_ty
...
    } else {
       compression = (double)100 - 100.0 * ((double)jcr->JobBytes / (double)jcr->ReadBytes);
-      if (compression < 0.5) {
-         bstrncpy(compress, "None", sizeof(compress));
-      } else {
-         Bsnprintf(compress, sizeof(compress), "%.1f %%", compression);
-         FindUsedCompressalgos(&compress_algo_list, jcr);
+      if(compression < 0){
+        Jmsg(jcr, M_WARNING, 0, _("Negative compression rate detected! "
+              "Disable compression for this fileset or choose different algorithm.\n"));
```
Comment (sebsura) [inline core/src/dird/backup.cc:954]: I would advise against this warning.  Think about the case were you do an incremental after only very few changes to incompressible files (for example you just updated some system libraries).  This probably trigger this warning even though the full fileset (which in this example is the whole filesystem) might be very compressible.

Comment (pstorz) [reply core/src/dird/backup.cc]: I have changed this in the last commit, so that we only get an INFO if a full backup expanded by more than 1%

**Review by sebsura — APPROVED** [review]
Looks good to me.  I was bitten by this already so i appreciate this change.

**Comment by arogge** [comment]
Can't we fix the underlying problem? i.e. don't write the compressed record if it is bigger than the uncompressed one. And maybe keep a statistic how often this happens and then skip further record compression if the file seems to be incompressible?

**Comment by sebsura** [comment]
That should be doable.

File: core/src/dird/backup.cc [hunk 2]
```diff
@@ -945,16 +945,18 @@ void GenerateBackupSummary(JobControlRecord *jcr, ClientDbRecord *cr, int msg_ty
...
-      } else {
-         Bsnprintf(compress, sizeof(compress), "%.1f %%", compression);
-         FindUsedCompressalgos(&compress_algo_list, jcr);
-      }
+     double compression = 100.0 - 100.0 * ((double)jcr->JobBytes / (double)jcr->ReadBytes);
+     if (compression < -1 && jcr->is_JobLevel(L_FULL)) {
+       Jmsg(jcr, M_INFO, 0,
+	    _("Compression inflated the full backup data by more than 1%%."
+	      " We suggest to use a different algorithm or to disable "
+	      "compression.\n"));
```
Comment (sebsura) [inline core/src/dird/backup.cc:957]: This needs to be fixed for it to be mergable.
```suggestion
	    T_("Compression inflated the full backup data by more than 1%%."
	       " We suggest to use a different algorithm or to disable "
	       "compression.\n"));
```

**Comment by arogge** [comment]
Builds fails as this still uses the old `_()` macro, so I added the nobuild flag for now.

**Pull request merged**